x-pack/plugins/observability_solution/dataset_quality/public/hooks/use_flyout_integration_actions.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getRouterLinkProps } from '@kbn/router-utils';
+import { useMemo, useCallback } from 'react';
+import useToggle from 'react-use/lib/useToggle';
+import { DashboardType } from '../../common/data_streams_stats';
+import { DASHBOARD_APP_LOCATOR, MANAGEMENT_APP_LOCATOR } from '../../common/constants';
+import { useKibanaContextForPlugin } from '../utils';
+
+export const useFlyoutIntegrationActions = () => {
+  const {
+    services: {
+      application: { navigateToUrl },
+      http: { basePath },
+      share,
+    },
+  } = useKibanaContextForPlugin();
+
+  const [isOpen, toggleIsOpen] = useToggle(false);
+
+  const dashboardLocator = useMemo(
+    () => share.url.locators.get(DASHBOARD_APP_LOCATOR),
+    [share.url.locators]
+  );
+  const indexManagementLocator = useMemo(
+    () => share.url.locators.get(MANAGEMENT_APP_LOCATOR),
+    [share.url.locators]
+  );
+
+  const handleCloseMenu = useCallback(() => {
+    toggleIsOpen();
+  }, [toggleIsOpen]);
+  const handleToggleMenu = useCallback(() => {
+    toggleIsOpen();
+  }, [toggleIsOpen]);
+
+  const getIntegrationOverviewLinkProps = useCallback(
+    (name: string, version: string) => {
+      const href = basePath.prepend(`/app/integrations/detail/${name}-${version}/overview`);
```

Comment:
Would be nice to have also a locator for integrations

Reply:
I think there is no one at the moment, or at least I couldn't find it, maybe not for this PR but we could introduce a locator for integrations in `packages/deeplinks`

Reply:
Maybe it's worth to ask integrations team whether they have one implemented or not, it would be awesome if we can rely on it